The klipper configuration of the Fly-Gemini motherboard suitable for v0

kluoyun opened the pull request:
<!--
Thank you for your interest in contributing to the VoronUsers repository, it is
highly appreciated!

**Please make sure the submission conforms to the rules outlined below. PRs which fail
to conform to the rules below are likely to be rejected.**
-->

  * [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folders names MUST NOT contain spaces. If possible, make sure file names also 
    do not contain any spaces.
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.
  * [x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

<!--
Describe the submission further using the template provided below. The more 
details the better!
-->

#### Which mods/configurations/profiles are added by this PR?

Fly-Gemini motherboard is used for v0 klipper configuration, fixing bracket, wiring diagram

#### How was it tested? 

It has been successfully tested on v0.1, and after several days of printing

#### Any background context you want to provide?

a 3d printer enthusiast

#### Screenshots (if appropriate)

![屏幕截图 2021-12-23 105131](https://user-images.githubusercontent.com/31061246/147181411-7549a2b6-e635-43dc-936f-b1f5e3dce76f.png)

#### Further notes

no

cbc02009 commented:
This is really great! My overall feedback is that I would like to see it fit the form of the standard v0 config to improve readability. I think it would be a good idea to start with the current v0 config, change the pins/settings around, and add in any extra stuff required for the Gemini with proper comments and such.

kluoyun commented:
> This is really great! My overall feedback is that I would like to see it fit the form of the standard v0 config to improve readability. I think it would be a good idea to start with the current v0 config, change the pins/settings around, and add in any extra stuff required for the Gemini with proper comments and such.

I added some comments and changed the configuration format

cbc02009 commented on firmware_configurations/README.md:
```diff
@@ -30,6 +30,7 @@ like so:
 | jaeger | [FYSETC Spider on V1.8](./klipper/jaeger/Spider_1.1_V1.8) | Example configuration and wiring diagram for a FYSETC Spider 1.1 board on the V1.8 | :x: | :heavy_check_mark: | :x: | :x: |
 | MarcPot | [Fysetc Spider Board Pins](./klipper/MarcPot/fysetc_spider_pins) | Defines a set of `[board_pins]` aliases for use with the Fysetc Spider controller | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
 | Mellow | [Fly-Super8 on V2.4](./klipper/Mellow/FLY-Super8) | Example configuration and wiring diagram for a FLY-Super8 board on the V2.4 | :x: | :x: | :heavy_check_mark: | :x: |
+| Mellow | [Fly-Gemini on v0](./klipper/Mellow/FLY-Gemini) | Sample configuration and wiring diagram of Fly-Gemini Linux integrated motherboard in v0 and v0.1 | :heavy_check_mark: | :x: | :x: | :x: |
```
please add an :x: for the VT column.

kluoyun replied:
VT col?

cbc02009 commented on firmware_configurations/klipper/Mellow/FLY-Gemini/cfg/printer.cfg:
```diff
@@ -0,0 +1,345 @@
+#####################################################################
+
+[stepper_x]
+##	Connected to X on mcu_xye (B Motor)
```
v0 doesn't have dual boards, so I would remove the `on mcu_xye` part

cbc02009 commented on firmware_configurations/klipper/Mellow/FLY-Gemini/cfg/printer.cfg:
```diff
@@ -0,0 +1,345 @@
+heater_pin: PA0
+sensor_pin: PC0
+##	Validate the following thermistor type to make sure it is correct
+sensor_type: PT1000
```
Please comment out the `sensor_type` line to force users to validate they have the correct sensor selected. This is for safety. also please add `##  See https://www.klipper3d.org/Config_Reference.html#common-thermistors for additional options` between 172 and 173 so the user can find the list of thermistor types.

cbc02009 commented on firmware_configurations/klipper/Mellow/FLY-Gemini/cfg/printer.cfg:
```diff
@@ -0,0 +1,345 @@
+heater_pin: PA2
+sensor_pin: PC2
+sensor_type: ATC Semitec 104GT-2
+control: watermark
```
any particular reason you're using watermark instead of PID for the bed heater?

cbc02009 commented on firmware_configurations/klipper/Mellow/FLY-Gemini/cfg/printer.cfg:
```diff
@@ -0,0 +1,345 @@
+[heater_bed]
+heater_pin: PA2
+sensor_pin: PC2
+sensor_type: ATC Semitec 104GT-2
```
Comment out this sensor for the same reasons and add the same comments as for the extruder heater.

cbc02009 commented on firmware_configurations/klipper/Mellow/FLY-Gemini/cfg/printer.cfg:
```diff
@@ -0,0 +1,345 @@
+#cycle_time: 0.50
+
+[fan_generic  Core_Fan]
+pin: host:gpiochip0/gpio3
```
Is this correct? I've never seen a forward slash used in pin assignments before.

kluoyun replied:
https://www.klipper3d.org/RPi_microcontroller.html#optional-identify-the-correct-gpiochip

cbc02009 reviewed (changes requested):
Please fix the orientation of the Gemini bracket: 
![image](https://user-images.githubusercontent.com/39525900/147609594-501a4185-526f-49d7-a36d-59f9f077441e.png)

cbc02009 commented on firmware_configurations/README.md:
```diff
@@ -30,6 +30,7 @@ like so:
 | jaeger | [FYSETC Spider on V1.8](./klipper/jaeger/Spider_1.1_V1.8) | Example configuration and wiring diagram for a FYSETC Spider 1.1 board on the V1.8 | :x: | :heavy_check_mark: | :x: | :x: | :x: |
 | MarcPot | [Fysetc Spider Board Pins](./klipper/MarcPot/fysetc_spider_pins) | Defines a set of `[board_pins]` aliases for use with the Fysetc Spider controller | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :x: |
 | Mellow | [Fly-Super8 on V2.4](./klipper/Mellow/FLY-Super8) | Example configuration and wiring diagram for a FLY-Super8 board on the V2.4 | :x: | :x: | :heavy_check_mark: | :x: | :x: |
+| Mellow | [Fly-Gemini on v0](./klipper/Mellow/FLY-Gemini) | Sample configuration and wiring diagram of Fly-Gemini Linux integrated motherboard in v0 and v0.1 | :heavy_check_mark: | :x: | :x: | :x: | :x: |
```
Just one more quick question. Are you an official representative for Mellow? if not this should be under your username instead of `Mellow`

kluoyun replied:
Yes I am

cbc02009 commented on firmware_configurations/klipper/Mellow/FLY-Gemini/cfg/printer.cfg:
```diff
@@ -0,0 +1,351 @@
+   M83                            ; set extruder to relative
+   G1 E10 F300                    ; extrude a little to soften tip
+   G1 E-40 F1800                  ; retract some, but not too much or it will jam
+   M82                 
```
please add an empty line to the end of the file

cbc02009 commented on printer_mods/README.md:
```diff
@@ -226,6 +226,7 @@ like so:
 | | [V0 TopHat Mini 12864](./Maverick_/V0_TopHat_Mini_12864) | Modified TopHat to mount a Mini 12864. | :heavy_check_mark: | :x: | :x: |:x:|
 | MCKURT |[Switchwire Y Belt Tensioner](./Mckurt/Switchwire_Y_Belt_Tensioner) | Switchwie Y axis Belt Tensioner | :x: | :x: | :x: | :heavy_check_mark: | :x: |
 | MCMBen | [Voron0 Block & Tackle Z Belt](./MCMBen/Voron0_Block_and_Tackle_Z_Belt) | A block & tackle belted Z solution using NEMA14. Ratios from 5:1 to 8:1 and friction prevent Z drop when powered off. | :heavy_check_mark: | :x: | :x: | :x: | :x: |
+| Mellow | [V0 Fly-Gemini](./Mellow/V0_Fly-Gemini) | Fly-Gemini adapted model for V0 | :heavy_check_mark: | :x: | :x: | :x: | :x:|
```
Same here. this should be under your username unless you're part of the Mellow team.

cbc02009 commented on printer_mods/Mellow/V0_Fly-Gemini/readme.md:
```diff
@@ -0,0 +1,4 @@
+# V0 adaptation model of Fly-Gemini
+
+## Fixed bracket
```
Please add a Parts BOM including what screws are needed to use the mounts and how many.

Other instalation instructions would also be useful. does this attach with VHB? Does the user need to drill holes for screws?

cbc02009 commented on printer_mods/Mellow/V0_Fly-Gemini/readme.md:
```diff
@@ -0,0 +1,31 @@
+### Gemini_v0.1_Stent_FLYLOGO.STL
+
+- Use VHB tape to secure to the V0 backplane
+- The bracket and the motherboard are directly connected with 4 M3*6 or M3*8 screws
```
use an x instead of *. The asterisk confuses the markup language. `M3x6 or M3x8`

kluoyun replied:
I was negligent, thanks

cbc02009 reviewed (changes requested):
Looks good! One quick change to the readme and it should be all set!
